packages/react-component-library/src/components/Timeline/Timeline.tsx

```diff
@@ -3,23 +3,67 @@ import React from 'react'
 import { TimelineProvider } from './context'
 
 import {
+  TimelineRootComponent,
+  TimelineHeadComponent,
+  TimelineBodyComponent,
+} from './types'
+
+import {
+  TimelineSide,
   TimelineDays,
+  TimelineWeeks,
   TimelineMonths,
-  TimelineRowsProps,
-  TimelineSide,
   TimelineTodayMarker,
-  TimelineWeeks,
+  TimelineRows,
 } from '.'
+
 import { TimelineOptions } from './context/types'
 import { DEFAULTS } from './constants'
 
 export interface TimelineProps extends ComponentWithClass {
-  children: React.ReactElement<TimelineRowsProps>
-  dayWidth?: number,
+  children: React.ReactElement | React.ReactElement[]
```

Comment:
It feels like this could be tighter as we know what types of subcomponents will be `children`